### ISSUE TITLE
BUG: New debs are made with tar.zst (need to revert to ubuntu-20.04 or ubuntu-18.04)

### ISSUE THREAD

**Issue by theofficialgman** [dave-theunsub/clamtk#152]
ubuntu-latest is now ubuntu 22.04

ubuntu has an extension to apt that uses tar.zst to compress file in debs. this is used by default by dpkg when packaging debs in 22.04

this prevents all debs from installing on debian/raspbian

you need to switch to building with ubuntu-20.04 or ubuntu-18.04 which do not use this functionality when packaging debs

https://github.com/Botspot/pi-apps/actions/runs/4738073287/jobs/8411536446#step:4:1613

alternativly you can specify the compression method as gzip through a dpkg command like option, debhelper override, or whatever method you use to build these debs

**Comment by dave-theunsub** [comment]
Thanks @theofficialgman ,

I'll take a look this weekend.

respectfully,
Dave M

**Comment by theofficialgman** [comment]
you might find the information I posted in here helpful for getting a start on implementing the changes if you wish to continue using 22.04 to build
https://github.com/pychess/pychess/issues/2049#issuecomment-1514927747

**Comment by dave-theunsub** [comment]
Hi @theofficialgman ,

Just to verify: building with 20.04.6 LTS (https://releases.ubuntu.com/20.04.6/) will do the trick?  Any particular naming scheme I need to follow when building that you can think of?

respectfully,  
Dave M

**Comment by theofficialgman** [comment]
yes simply changing to 20.04 will work since tar.xz was still the default for dpkg building there. no other changes necessary.

depending on how you make your debs it might be simple enough to keep using 22.04 though and just specify the compression type that you want. I don't know how you are making your debs (what tool) so can't really give you more info than that. refer to the linked comment in my previous post for more.

**Comment by dave-theunsub** [comment]
Hi @theofficialgman ,

Care to test it out?  Hashes have been updated as well:
https://github.com/dave-theunsub/clamtk/releases/tag/v6.15

respectfully,  
Dave M

**Comment by theofficialgman** [comment]
I see the _2004 deb

I don't think you understood what I was asking. Don't make two separate debs. Just use 20.04 to build the one deb and keep the name `clamtk_6.15-1_all.deb`. the contents of the archives in the deb are identical no matter what you package on since perl is not a compiled language.
